app/src/main/java/fi/helsinki/opintoni/integration/publicwww/PublicWwwRestClient.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of MystudiesMyteaching application.
+ *
+ * MystudiesMyteaching application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MystudiesMyteaching application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MystudiesMyteaching application.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package fi.helsinki.opintoni.integration.publicwww;
+
+import com.rometools.rome.feed.atom.Feed;
+import fi.helsinki.opintoni.config.AppConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class PublicWwwRestClient {
+    private final String baseUrl;
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    private AppConfiguration appConfiguration;
+
+    public PublicWwwRestClient(String baseUrl, RestTemplate restTemplate) {
```

Comment:
Kannattaa tehdä rakentajasta `@Autowired` joka saa paramina `AppConfiguration`in. `baseUrl`in saa suoraan `appConfiguration`illa ja `restTemplate`n voi rakentaa myös private-metodikutsun kautta rakentajassa.